java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java

```diff
@@ -47,12 +47,14 @@ public static CdcEngineRunner newCdcEngineRunner(
                             config.getResolvedDebeziumProps(),
                             (success, message, error) -> {
                                 if (!success) {
-                                    responseObserver.onError(error);
                                     LOG.error(
                                             "engine#{} terminated with error. message: {}",
                                             sourceId,
                                             message,
                                             error);
+                                    if (error != null) {
+                                        responseObserver.onError(error);
+                                    }
```

Comment:
Any motivation for this change? In what situation will it failed with `error == null`?

Reply:
For defensive of nullptr and match the implementation of the following `create(DbzConnectorConfig config, long channelPtr)` method.
Actually `newCdcEngineRunner` only used in unit test right now, since we replaced with embeded jvm.